lib/eslint.js

```diff
@@ -173,6 +173,18 @@ function addDeclaredGlobals(program, globalScope, config) {
             variable.eslintUsed = true;
         }
     });
+
```

Comment:
Can you add a comment here explaining what's happening? I think something like this:

> "through" contains all variables that are referenced but their definition cannot be found. Since we augment the global scope using configuration, we need to update references and remove the ones that were added by configuration.

Also,, can you add a test for this?